### ISSUE TITLE
Failed mounting in archiveloop, after one step flash

### ISSUE THREAD

**Issue by seandoesgis** [marcone/teslausb#42]
```
Sun 26 May 18:16:38 BST 2019: Mounting /mnt/archive...
mount error(22): Invalid argument
Refer to the mount.cifs(8) manual page (e.g. man mount.cifs)
Sun 26 May 18:16:38 BST 2019: Failed to mount /mnt/archive.
Sun 26 May 18:16:38 BST 2019: Sleeping before retry...
```
 How would I find the invalid argument?  Only thing I've been able to find online is the CIFS version may need to be changed to something like 2.1.  How would I be able to try that.

**Comment by myquealer** [comment]
Make sure the folder you're mounting has no spaces or other special characters in its name. This tripped me up for a while, and I think this is the error I was getting.

The documentation should make this requirement clear, or just surround each mounting argument in quotes so it works with spaces.

**Comment by marcone** [comment]
Does /mnt/archive exist?
Does it appear in /etc/fstab?
Are the username and password in /root/.teslaCamArchiveCredentials correct?

**Comment by seandoesgis** [comment]
> Does /mnt/archive exist?
> Does it appear in /etc/fstab?
> Are the username and password in /root/.teslaCamArchiveCredentials correct?

```
root@teslausb:~# cd /mnt/archive
root@teslausb:/mnt/archive# ls -a
.  ..
root@teslausb:/mnt/archive# cd /etc/fstab
-bash: cd: /etc/fstab: Not a directory
root@teslausb:/mnt/archive# 
```
There are video files on the pi though when I mount to my computer
Credentials are fine in .teslaCamArchiveCredentials...I think.  I'm on MacOS and setup a shared folder and created a user named TeslaCam with a password and gave that user read/write to the folder.  Maybe there is some trick I don't know with the file sharing in MacOS

**Comment by marcone** [comment]
/etc/fstab is a file, not a directory. Check the content of that file and see if there's an entry for /mnt/archive in there

**Comment by seandoesgis** [comment]
> /etc/fstab is a file, not a directory. Check the content of that file and see if there's an entry for /mnt/archive in there

`//192.168.1.20/TeslaCam /mnt/archive cifs vers=3.0,credentials=/root/.teslaCamArchiveCredentials,iocharset=utf8,file_mode=0777,dir_mode=0777 0`

**Comment by marcone** [comment]
That looks fine. You say your share is on a Mac though. Another user had problems with shared drive on MacOS too (see https://github.com/marcone/teslausb/issues/27), and while it should in theory be as easy as telling the Mac to use SMB instead of AFP, I couldn't get it to work myself either.

**Comment by seandoesgis** [comment]
> That looks fine. You say your share is on a Mac though. Another user had problems with shared drive on MacOS too (see #27), and while it should in theory be as easy as telling the Mac to use SMB instead of AFP, I couldn't get it to work myself either.

Doh...thanks marcone, I'll have to try rclone4pi or something.

**Comment by myquealer** [comment]
I set up TeslaUSB with a Mac and it is saving to a Mac. I lowered gradually  CIFS version to 1.0, and then realized other problems were preventing it from working. It looks like you're still on CIFS 3.0. You might try 2.1, 2.0 or 1.0 and see if it works for you. I'm still on 1.0, since it's working for me.

**Comment by seandoesgis** [comment]
> I set up TeslaUSB with a Mac and it is saving to a Mac. I lowered gradually CIFS version to 1.0, and then realized other problems were preventing it from working. It looks like you're still on CIFS 3.0. You might try 2.1, 2.0 or 1.0 and see if it works for you. I'm still on 1.0, since it's working for me.

I actually got it working today on the Mac with the default CIFS version 3.0.  When entering in variables for the config I simply used no quotes for the password or others.  I also made sure the folder didn't have cap characters.

`export ARCHIVE_SYSTEM=cifs
export archiveserver=192.168.1.20
export sharename=teslacam
export shareuser=ssssss
export sharepassword=xxxxxx
export camsize=95%
export SSID=p00p
export WIFIPASS='111111111'
export HEADLESS_SETUP=true
export timezone=America/New_York
`
